### PR TITLE
API Remove deprecated code

### PR DESCRIPTION
**Pull request by sabina-talipova**
### Parent issue
- https://github.com/silverstripe/silverstripe-framework/issues/10501

### CODE REVIEW

File: lang/ar.yml [hunk 1]
```diff
@@ -167,8 +167,6 @@ ar:
     many_many_BackLinkTracking: 'متابعة التعقيبات'
     many_many_ImageTracking: 'متابعة الصور'
     many_many_LinkTracking: 'متابعة الروابط'
-  SilverStripe\CMS\Model\SiteTreeFileExtension:
```
Comment (emteknetnz) [inline lang/ar.yml]: Can you please restore ALL the translation updates - we cannot delete keys from the yml files.  If we do, when we push the changes to transifex the translations will be deleted from transifex, so if the translations are pulled for older versions of silverstripe, then the translations will be deleted from the yml files of the older version of silverstripe

File: tests/php/Model/SiteTreeTest.php [hunk 2]
```diff
@@ -888,7 +887,7 @@ public function testAuthorIDAndPublisherIDFilledOutOnPublish()
             "SELECT \"AuthorID\", \"PublisherID\" FROM \"SiteTree_Versions\"
             WHERE \"RecordID\" = ? ORDER BY \"Version\" DESC",
             [$about->ID]
-        )->first();
+        )->record();
```
Comment (emteknetnz) [inline tests/php/Model/SiteTreeTest.php:890]: I'm surprised this used to work

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: tests/php/Model/SiteTreeTest.php (hunk 2 above)
Comment (sabina-talipova) [reply tests/php/Model/SiteTreeTest.php:890]: Should I return back the method `first()`?

File: lang/ar.yml (hunk 1 above)
Comment (sabina-talipova) [reply lang/ar.yml]: Done

File: tests/php/Model/SiteTreeTest.php (hunk 2 above)
Comment (emteknetnz) [reply tests/php/Model/SiteTreeTest.php:890]: No it's correct, first() was just a deprecated alias for record() - sorry I should have checked that before commenting

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**